From f12ffe4d775e33bcabd4e1f5171ee08964429379 Mon Sep 17 00:00:00 2001
From: asalde_le1 <57564314+asaldele1@users.noreply.github.com>
Date: Fri, 24 May 2024 22:18:04 +0000
Subject: [PATCH 1/4] Revert "Native: ProducerListener: Import
 ON_BUFFER_DETACHED Changes"

This reverts commit 53b85c2dc4d7af2de83833fa3b7424f73501fd43.

Signed-off-by: asalde_le1 <57564314+asaldele1@users.noreply.github.com>
---
 libs/gui/IProducerListener.cpp                | 21 -------------------
 .../gui/bufferqueue/1.0/WProducerListener.cpp |  3 ---
 .../bufferqueue/2.0/H2BProducerListener.cpp   |  3 ---
 libs/gui/include/gui/IProducerListener.h      |  2 --
 libs/gui/include/gui/Surface.h                | 10 ---------
 .../gui/bufferqueue/1.0/WProducerListener.h   |  1 -
 .../gui/bufferqueue/2.0/H2BProducerListener.h |  1 -
 7 files changed, 41 deletions(-)

diff --git a/libs/gui/IProducerListener.cpp b/libs/gui/IProducerListener.cpp
index 72cec7afad..0683087211 100644
--- a/libs/gui/IProducerListener.cpp
+++ b/libs/gui/IProducerListener.cpp
@@ -25,7 +25,6 @@ enum {
     ON_BUFFER_RELEASED = IBinder::FIRST_CALL_TRANSACTION,
     NEEDS_RELEASE_NOTIFY,
     ON_BUFFERS_DISCARDED,
-    ON_BUFFER_DETACHED,
 };
 
 class BpProducerListener : public BpInterface<IProducerListener>
@@ -65,13 +64,6 @@ public:
         data.writeInt32Vector(discardedSlots);
         remote()->transact(ON_BUFFERS_DISCARDED, data, &reply, IBinder::FLAG_ONEWAY);
     }
-
-    virtual void onBufferDetached(int slot) {
-        Parcel data, reply;
-        data.writeInterfaceToken(IProducerListener::getInterfaceDescriptor());
-        data.writeInt32(slot);
-        remote()->transact(ON_BUFFER_DETACHED, data, &reply, IBinder::FLAG_ONEWAY);
-    }
 };
 
 // Out-of-line virtual method definition to trigger vtable emission in this
@@ -96,10 +88,6 @@ public:
     virtual void onBuffersDiscarded(const std::vector<int32_t>& discardedSlots) override {
         return mBase->onBuffersDiscarded(discardedSlots);
     }
-
-    virtual void onBufferDetached(int slot) {
-        mBase->onBufferDetached(slot);
-    }
 };
 
 IMPLEMENT_HYBRID_META_INTERFACE(ProducerListener,
@@ -127,12 +115,6 @@ status_t BnProducerListener::onTransact(uint32_t code, const Parcel& data,
             onBuffersDiscarded(discardedSlots);
             return NO_ERROR;
         }
-        case ON_BUFFER_DETACHED:
-            int slot = 0;
-            CHECK_INTERFACE(IProducerListener, data, reply);
-            data.readInt32(&slot);
-            onBufferDetached(slot);
-            return NO_ERROR;
     }
     return BBinder::onTransact(code, data, reply, flags);
 }
@@ -146,7 +128,4 @@ bool BnProducerListener::needsReleaseNotify() {
 void BnProducerListener::onBuffersDiscarded(const std::vector<int32_t>& /*discardedSlots*/) {
 }
 
-void BnProducerListener::onBufferDetached(int slot) {
-    ALOGE("BnProducerListener::onBufferDetached slot: %d",slot);
-}
 } // namespace android
diff --git a/libs/gui/bufferqueue/1.0/WProducerListener.cpp b/libs/gui/bufferqueue/1.0/WProducerListener.cpp
index 7a3e050561..56b64b9ddd 100644
--- a/libs/gui/bufferqueue/1.0/WProducerListener.cpp
+++ b/libs/gui/bufferqueue/1.0/WProducerListener.cpp
@@ -49,7 +49,4 @@ bool LWProducerListener::needsReleaseNotify() {
 void LWProducerListener::onBuffersDiscarded(const std::vector<int32_t>& /*slots*/) {
 }
 
-void LWProducerListener::onBufferDetached(int /*slot*/) {
-}
-
 }  // namespace android
diff --git a/libs/gui/bufferqueue/2.0/H2BProducerListener.cpp b/libs/gui/bufferqueue/2.0/H2BProducerListener.cpp
index 745ffea75f..b2bd1172d6 100644
--- a/libs/gui/bufferqueue/2.0/H2BProducerListener.cpp
+++ b/libs/gui/bufferqueue/2.0/H2BProducerListener.cpp
@@ -55,9 +55,6 @@ bool H2BProducerListener::needsReleaseNotify() {
 void H2BProducerListener::onBuffersDiscarded(const std::vector<int32_t>& /*slots*/) {
 }
 
-void H2BProducerListener::onBufferDetached(int /*slot*/) {
-}
-
 }  // namespace utils
 }  // namespace V2_0
 }  // namespace bufferqueue
diff --git a/libs/gui/include/gui/IProducerListener.h b/libs/gui/include/gui/IProducerListener.h
index d2fa2c8981..b15f501518 100644
--- a/libs/gui/include/gui/IProducerListener.h
+++ b/libs/gui/include/gui/IProducerListener.h
@@ -78,7 +78,6 @@ public:
             Parcel* reply, uint32_t flags = 0);
     virtual bool needsReleaseNotify();
     virtual void onBuffersDiscarded(const std::vector<int32_t>& slots);
-    virtual void onBufferDetached(int slot);
 };
 
 #else
@@ -92,7 +91,6 @@ public:
     virtual ~StubProducerListener();
     virtual void onBufferReleased() {}
     virtual bool needsReleaseNotify() { return false; }
-    virtual void onBufferDetached(int /**slot**/) {}
 };
 
 } // namespace android
diff --git a/libs/gui/include/gui/Surface.h b/libs/gui/include/gui/Surface.h
index ffb04bb54c..4ea1e0c843 100644
--- a/libs/gui/include/gui/Surface.h
+++ b/libs/gui/include/gui/Surface.h
@@ -56,11 +56,6 @@ public:
     virtual bool needsReleaseNotify() = 0;
 
     virtual void onBuffersDiscarded(const std::vector<sp<GraphicBuffer>>& buffers) = 0;
-
-    virtual void onBufferDetached(int /**slot**/) {
-        //default do nothing
-    }
-
 };
 
 /*
@@ -419,11 +414,6 @@ protected:
         }
 
         virtual void onBuffersDiscarded(const std::vector<int32_t>& slots);
-
-        virtual void onBufferDetached(int slot) {
-            mSurfaceListener->onBufferDetached(slot);
-        }
-
     private:
         wp<Surface> mParent;
         sp<SurfaceListener> mSurfaceListener;
diff --git a/libs/gui/include/gui/bufferqueue/1.0/WProducerListener.h b/libs/gui/include/gui/bufferqueue/1.0/WProducerListener.h
index efdd5aad5a..197db26444 100644
--- a/libs/gui/include/gui/bufferqueue/1.0/WProducerListener.h
+++ b/libs/gui/include/gui/bufferqueue/1.0/WProducerListener.h
@@ -55,7 +55,6 @@ public:
     void onBufferReleased() override;
     bool needsReleaseNotify() override;
     void onBuffersDiscarded(const std::vector<int32_t>& slots) override;
-    void onBufferDetached(int slot) override;
 };
 
 }  // namespace android
diff --git a/libs/gui/include/gui/bufferqueue/2.0/H2BProducerListener.h b/libs/gui/include/gui/bufferqueue/2.0/H2BProducerListener.h
index d66c025fe3..92650b701b 100644
--- a/libs/gui/include/gui/bufferqueue/2.0/H2BProducerListener.h
+++ b/libs/gui/include/gui/bufferqueue/2.0/H2BProducerListener.h
@@ -47,7 +47,6 @@ public:
     virtual void onBufferReleased() override;
     virtual bool needsReleaseNotify() override;
     virtual void onBuffersDiscarded(const std::vector<int32_t>& slots) override;
-    virtual void onBufferDetached(int slot) override;
 };
 
 }  // namespace utils
-- 
2.30.2

